clarity/src/vm/tests/variables.rs

```diff
@@ -221,3 +221,466 @@ fn test_tenure_height(
+        (define-private (test-func)
+            (map-insert block-height u1 u2)
+        )
+        "#,
+        &[
+            (
+                WhenError::Initialization,
+                |version, _| version < ClarityVersion::Clarity3,
+                CheckErrors::NameAlreadyUsed("block-height".to_string()),
+            ),
+            (
+                WhenError::Analysis,
+                |version, _| version >= ClarityVersion::Clarity3,
+                CheckErrors::ReservedWord("block-height".to_string()),
+            ),
+        ],
+        Value::Bool(true),
+    );
+
+    // let
+    expect_contract_error(
+        version,
+        epoch,
+        &mut tl_env_factory,
+        "let",
+        r#"
+        (define-private (test-func)
+            (let ((block-height 32))
+                block-height
+            )
+        )
+        "#,
+        &[
+            (
+                WhenError::Runtime,
+                |version, _| version < ClarityVersion::Clarity3,
+                CheckErrors::NameAlreadyUsed("block-height".to_string()),
+            ),
+            (
+                WhenError::Analysis,
+                |version, _| version >= ClarityVersion::Clarity3,
+                CheckErrors::ReservedWord("block-height".to_string()),
+            ),
+        ],
+        Value::Int(32),
+    );
+
+    // match binding
+    expect_contract_error(
+        version,
+        epoch,
+        &mut tl_env_factory,
+        "match-binding",
+        r#"
+        (define-read-only (test-func)
+          (let ((x (if true (ok u5) (err u7))))
+            (match x
+              block-height 3
+              e 4
+            )
+          )
+        )
+        "#,
+        &[
+            (
+                WhenError::Runtime,
+                |version, _| version < ClarityVersion::Clarity3,
+                CheckErrors::NameAlreadyUsed("block-height".to_string()),
+            ),
+            (
+                WhenError::Analysis,
+                |version, _| version >= ClarityVersion::Clarity3,
+                CheckErrors::ReservedWord("block-height".to_string()),
+            ),
+        ],
+        Value::Int(3),
+    );
+
+    // function
+    expect_contract_error(
+        version,
+        epoch,
+        &mut tl_env_factory,
+        "function",
+        r#"
+        (define-private (block-height) true)
+        (define-private (test-func) (block-height))
+        "#,
+        &[
+            (
+                WhenError::Initialization,
+                |version, _| version < ClarityVersion::Clarity3,
+                CheckErrors::NameAlreadyUsed("block-height".to_string()),
+            ),
+            (
+                WhenError::Analysis,
+                |version, _| version >= ClarityVersion::Clarity3,
+                CheckErrors::ReservedWord("block-height".to_string()),
+            ),
+        ],
+        Value::Bool(true),
+    );
+}
+
+#[apply(test_clarity_versions)]
+fn reuse_stacks_block_height(
```

Comment:
Same concerns here -- there are other Clarity built-ins that let you create named data that need to be tested.

Reply:
Added in 890f07cd1d2b939f9f9daa0f497824e7b4359ff7.